selfservice/hook/web_hook.go

```diff
@@ -499,3 +501,18 @@ func isTimeoutError(err error) bool {
 	var te interface{ Timeout() bool }
 	return errors.As(err, &te) && te.Timeout() || errors.Is(err, context.DeadlineExceeded)
 }
+
+func instrumentHTTPClientForEvents(ctx context.Context, httpClient *retryablehttp.Client) {
+	var (
+		attempt = 0
+		reqBody []byte
+	)
+	httpClient.RequestLogHook = func(_ retryablehttp.Logger, req *http.Request, retryNumber int) {
+		attempt = retryNumber + 1
+		reqBody, _ = httputil.DumpRequestOut(req, true)
+	}
+	httpClient.ResponseLogHook = func(_ retryablehttp.Logger, res *http.Response) {
+		resBody, _ := httputil.DumpResponse(res, true)
+		trace.SpanFromContext(ctx).AddEvent(events.NewWebhookDelivered(ctx, res.Request.URL, reqBody, res.StatusCode, resBody, attempt))
```

Comment:
Both `reqBody` and `resBody` can be arbitrarily long. Should we limit this somehow? IIUC, if the event attributes are too large, the event is just dropped. But such events should still show up, right?

Reply:
Very good point!

Reply:
Added a limit for resBody. For reqBody, I want to add a limit to `jsonnetsecure` instead. That's a DOS vector anyway and needs to be fixed there.